validation/mock_trace_validation.go

```diff
@@ -0,0 +1,68 @@
+package validation
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/golang/protobuf/ptypes"
+	"google.golang.org/genproto/googleapis/devtools/cloudtrace/v2"
+)
+
+const (
+	InvalidTimestampMsg   = "start time must be before end time"
```

Comment:
Why are these exported?

If callers need to check the error type then export an error constant. e.g. https://golang.org/pkg/os/exec/#pkg-variables

Reply:
Got it, I'll export the errors instead.